Libs/libMacGitverCore/Widgets/ProgressDlg.cpp

```diff
@@ -0,0 +1,206 @@
+
+#include "ProgressDlg.hpp"
+#include "ui_ProgressDlg.h"
+#include "ui_ProgressWdgt.h"
+
+#include <QStringBuilder>
+#include <QCloseEvent>
+#include <QPushButton>
+
+namespace Private
+{
+
+    class ProgressWdgt : public QWidget, public Ui::ProgressWdgt
+    {
+    public:
+        typedef QMap< QString, QPointer<ProgressWdgt> > Steps;
+
+        enum Status { Running = 0, Stopped };
+
+    public:
+        ProgressWdgt(const QString& description)
+        {
+            setupUi(this);
+            progressBar->setMinimum(0);
+            progressBar->setMaximum(100);
+            txtHeader->setText( description );
+        }
+
+    public:
+        Steps   mSteps;
+        Status  mStatus     = Running;
```

Comment:
Both, actually. If I found a constructor only initialising half of the members, I'd probably consider this an error and fix it - though there might not be a problem at all, considering that some member could be initialised in the class definition.

Technically spoken: At which point can you assume which variable to be initialised? We know the order changed from declaration-order to access-then-declaration-order from C++03 to C++11. Trouble starts when you have to use `this` inside a constructor's base-class or member-initialiser-list...